packages/puppeteer-core/src/common/NetworkManager.ts

```diff
@@ -209,11 +219,22 @@ export class NetworkManager extends EventEmitter {
       ? networkConditions.latency
       : 0;
 
-    await this.#updateNetworkConditions();
+    await this.#applyToAllClients(this.#applyNetworkConditions.bind(this));
+  }
+
+  async #applyToAllClients(fn: (client: CDPSession) => Promise<unknown>) {
+    await Promise.all(
+      Array.from(this.#clients).map(client => {
```

Comment:
```suggestion
      this.#clients.values().map(client => {
```

Reply:
Values does not return an array so has no map.